index.js

```diff
@@ -11,7 +12,15 @@ const resolveFrom = (fromDir, moduleId, silent) => {
 		throw new TypeError(`Expected \`moduleId\` to be of type \`string\`, got \`${typeof moduleId}\``);
 	}
 
-	fromDir = path.resolve(fromDir);
+	try {
+		fromDir = fs.realpathSync(fromDir);
+	} catch (err) {
+		if (err.code !== 'EINVAL') {
```

Comment:
Not really sure why you're just throwing on `EINVAL`. Any error should be thrown.

You also have to correctly handle the `silent` argument. See line 32.